uvicorn/lifespan/on.py

```diff
@@ -31,7 +49,7 @@ async def startup(self) -> None:
         # Keep a hard reference to prevent garbage collection
         # See https://github.com/encode/uvicorn/pull/972
 
-        await self.receive_queue.put({"type": "lifespan.startup"})
+        await self.receive_queue.put(LifespanStartupEvent(type="lifespan.startup"))
```

Comment:
Not doing this gives me: 
```bash
uvicorn/lifespan/on.py:52: error: Type of TypedDict is ambiguous, could be any of ("LifespanStartupEvent", "LifespanShutdownEvent")
uvicorn/lifespan/on.py:52: error: Argument 1 to "put" of "Queue" has incompatible type "Dict[str, str]"; expected "Union[LifespanStartupEvent, LifespanShutdownEvent]"
```
If you have suggestions, I'm open to change it :kissing:

Reply:
I think if you do something like 
```
lste : LifespanStartupEvent = {"type": "lifespan.startup"}
await self.receive_queue.put(lste)
```
it should be ok, though I have not tested it

Reply:
Yeah, it should work. But I don't see any real improvement. Is there any?

Reply:
> Yeah, it should work. But I don't see any real improvement. Is there any?

well not having a mypy error is one in my mind, I'm no mypy pro but found the TypedDict experience quite frustrating and solved most of my issues by doing that, I'm not married to it though so if you have a better idea I'm all ears.

Reply:
Well, the idea would be the current implementation on the PR, but it's the same to me. I've never done what I propose on this PR tho, I always do what you suggest, so I'm going to change the implementation.

Reply:
Done